operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -193,11 +180,27 @@ public Result<R> match(R actualResource, R desired, P primary, Context<P> contex
     }
   }
 
+  private void copySDKAnnotations(R actualResource, final R desired) {
+    String actual = actualResource.getMetadata().getAnnotations()
+        .get(InformerEventSource.PREVIOUS_ANNOTATION_KEY);
+    Map<String, String> annotations = desired.getMetadata().getAnnotations();
+    if (actual != null) {
+      if (annotations == null) {
```

Comment:
I don't think that we need that part as the client always make sure the annotations exist, if I'm not mistaken. Unless, of course, we want to safeguard here, regardless of what the client does.

Reply:
The client will initialize the annotation, but the desired is coming from the user logic and could have been set to null.  I was just trying to match a similar guard that is used in addDefaultSecondaryToPrimaryMapperAnnotations - but I'd be fine if the assumption was that the user must leave the annotations as non-null and remove the initialization from both of those places.

Also it may not need to be part of these changes, but it's making things more complicated than needed to diffentiate the ssa and non-ssa cases wrt methods like addReferenceHandlingMetadata - I think this should be refactored so that the update / match logic can read more like create so that we'll take the same pre-steps in both cases.

Reply:
@metacosm see what you think of latest commit